charms/jupyter-ui/src/charm.py

```diff
@@ -156,6 +146,38 @@ def set_pod_spec(self, event):
         )
         self.model.unit.status = ActiveStatus()
 
+    def _configure_mesh(self, interfaces):
+        if interfaces["ingress"]:
+            interfaces["ingress"].send_data(
+                {
+                    "prefix": "/",
+                    "rewrite": "/",
+                    "service": self.model.app.name,
+                    "port": self.model.config["port"],
+                }
+            )
+
+    def _check_leader(self):
+        if not self.unit.is_leader():
+            # We can't do anything useful when not the leader, so do nothing.
+            raise CheckFailed("Waiting for leadership", WaitingStatus)
+
+    def _get_interfaces(self):
+        try:
+            interfaces = get_interfaces(self)
+        except NoVersionsListed as err:
+            raise CheckFailed(err, WaitingStatus)
+        except NoCompatibleVersions as err:
+            raise CheckFailed(err, BlockedStatus)
```

Comment:
I had the same error as you in CI and was able to fix it by setting this code to the following:
```python
            raise CheckFailed(str(err), WaitingStatus)
        except NoCompatibleVersions as err:
            raise CheckFailed(str(err), BlockedStatus)
```

I'm not really sure why this fails only in some specifc charms and in others it doesn't. It shouldn't fail in the first please as these expections have both a `__str__` method.